### PR TITLE
Faster tests

### PR DESCRIPTION
**Pull request by sebastien-j**
@nouiz , @abergeron ,

Unfinished work.

I tried the changes proposed by @nouiz. Unfortunately, they did not make the tests faster.

### CODE REVIEW

File: theano/sandbox/cuda/tests/test_fftconv.py [hunk 1]
```diff
@@ -188,8 +188,12 @@ def run_conv_valid(self, inputs_shape, filters_shape, pad=False):
                                                           pad_last_dim = pad)
         conv_fft = conv_fft.dimshuffle(0, 2, 3, 4, 1)
 
-        f_ref = theano.function([], conv_ref)
-        f_fft = theano.function([], conv_fft, mode=mode_with_gpu)
+        ref_mode = theano.compile.get_default_mode()
+        ref_mode.check_py = False
+        f_ref = theano.function([], conv_ref, mode=ref_mode)
+        mode = mode_with_gpu
+        mode.check_py = False
```
Comment (nouiz) [inline theano/sandbox/cuda/tests/test_fftconv.py]: my mistake, "check_py -> check_py_code". idem for all place.

File: theano/sandbox/cuda/tests/test_gemmcorr3d.py [hunk 2]
```diff
@@ -172,8 +172,11 @@ def test_opt_conv3d_gemm(self):
         conv = theano.tensor.nnet.conv3D(V=inputs, W=filters,
                                          b=bias, d=(1, 1, 1))
         mode = mode_with_gpu.including('conv3d_gemm')
+        mode.check_py = False
+        ref_mode = theano.compile.get_default_mode()
```
Comment (nouiz) [inline theano/sandbox/cuda/tests/test_gemmcorr3d.py]: ref_mode = copy.copy(theano.compile.get_default_mode())
ref_mode.check_py_code = False

need "import copy"

**Comment by sebastien-j** [comment]
The fft tests are now faster in `DEBUG_MODE`. They are still slow with `FAST_COMPILE`.

`test_GPU_nstreams_limit()` is still slow in `DEBUG_MODE`.

**Comment by nouiz** [comment]
thanks. I'll merge it now before a new daily buildbot start.

**Comment by nouiz** [comment]
For the test test_opt_convgrad3d_fft, I saw that we compute the default value like this:
        f_ref = theano.function([], conv)
I think we can do:
        f_ref = theano.function([], conv, mode="FAST_RUN")

We do not need the reference result in python mode. This will force to use the c code, so it will be faster. Can you check the other slow tests in FAST_COMPILE, maybe the same change will do the work.

**Pull request merged**